pkg/prometheus/operator.go

```diff
@@ -138,6 +138,17 @@ func (cg *ConfigGenerator) validateRemoteWriteSpec(spec monitoringv1.RemoteWrite
 				return fmt.Errorf("the provided Azure OAuth clientId is invalid")
 			}
 		}
+
+		if spec.AzureAD.WorkloadIdentity != nil {
```

Comment:
we should also check the Prometheus version and return an error if not supported?